### ISSUE TITLE
Correct JavaScript code-generation for some/none 

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/ergo#445]
Currently, the code-generation for some/none generates a left/right record in JSON which isn't consistent with the CTO.

```
bash-3.2$ cat model.cto 
namespace org.accordproject.helloworldoption

transaction Request {
  o String input
}

transaction Response {
  o String output optional
}

@AccordTemplateModel("helloworld")
concept TemplateModel {
  /**
   * The name for the clause
   */
  o String name
}
bash-3.2$ cat logic.ergo 
namespace org.accordproject.helloworldoption

contract HelloWorldOption over TemplateModel {
  // Simple Clause
  clause sayhello(request : Request) : Response {
    return Response{ output: some("Hello " ++ contract.name ++ " (" ++ request.input ++ ")") }
  }
}
bash-3.2$ ergo execute --cto ./model.cto --ergo ./logic.ergo --contractname org.accordproject.helloworldoption.HelloWorldOption --contract ./contract.json --request ./request.json --state ./state.json
11:28:47 - info: Logging initialized. 2018-09-06T15:28:47.343Z
11:28:47 - info: {"response":{"output":{"left":"Hello Fred Blogs (Accord Project)"},"$class":"org.accordproject.helloworldoption.Response"},"state":{"$class":"org.accordproject.cicero.contract.AccordContractState","stateId":"1"},"emit":[]}
bash-3.2$ 
```

**Comment by jeromesimeon** [comment]
Note this issue is blocking for type-checking (since it forces use of optional constructors).

**Comment by jeromesimeon** [comment]
Addressed in PR #446 by using a custom translation from some/none operations in Ergo to NNRC which does not rely on sum types.
